packages/in_app_purchase/in_app_purchase_platform_interface/lib/src/in_app_purchase_platform.dart

```diff
@@ -0,0 +1,216 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+import 'types/types.dart';
+
+/// The interface that implementations of in_app_purchase must implement.
+///
+/// Platform implementations should extend this class rather than implement it as `in_app_purchase`
+/// does not consider newly added methods to be breaking changes. Extending this class
+/// (using `extends`) ensures that the subclass will get the default implementation, while
+/// platform implementations that `implements` this interface will be broken by newly added
+/// [InAppPurchasePlatform] methods.
+abstract class InAppPurchasePlatform extends PlatformInterface {
+  /// Constructs a InAppPurchasePlatform.
+  InAppPurchasePlatform() : super(token: _token);
+
+  static final Object _token = Object();
+
+  //// Ensures that implementers are using `extends` rather than
+  /// `implements` and throws [AssertionError] if not.
+  ///
+  /// This is implemented as a static method so that it cannot be overridden
+  /// with `noSuchMethod`.
+  static void verifyToken(InAppPurchasePlatform instance) {
```

Comment:
Hmm this is a hard question. Now how do we force implementers to call this method? :)

Reply:
Hahaha this also crossed my mind and I couldn't think of a solution (I was hoping that you'd have some brilliant idea to be honest ;) )

Reply:
@cyanglaz last week we discussed this issue offline and you had made some changes allowing a nullable `instance`. Could you share these changes so I can add them to this PR?

Reply:
Updated with the code discussed offline